core-to-plc/src/Language/Plutus/CoreToPLC.hs

```diff
@@ -451,41 +451,41 @@ isPrimitiveDataCon dc = dc == GHC.intDataCon
 -- These never seem to come up, rather we get the typeclass operations. Not sure if we need them.
 convPrimitiveOp :: (Converting m) => GHC.PrimOp -> m PLCExpr
 convPrimitiveOp = \case
-    GHC.IntAddOp  -> mkIntFun PLC.AddInteger
-    GHC.IntSubOp  -> mkIntFun PLC.SubtractInteger
-    GHC.IntMulOp  -> mkIntFun PLC.MultiplyInteger
+    GHC.IntAddOp  -> pure $ mkIntFun PLC.AddInteger
+    GHC.IntSubOp  -> pure $ mkIntFun PLC.SubtractInteger
+    GHC.IntMulOp  -> pure $ mkIntFun PLC.MultiplyInteger
     -- check this one
-    GHC.IntQuotOp -> mkIntFun PLC.DivideInteger
-    GHC.IntRemOp  -> mkIntFun PLC.RemainderInteger
-    GHC.IntGtOp   -> mkIntRel PLC.GreaterThanInteger
-    GHC.IntGeOp   -> mkIntRel PLC.GreaterThanEqInteger
-    GHC.IntLtOp   -> mkIntRel PLC.LessThanInteger
-    GHC.IntLeOp   -> mkIntRel PLC.LessThanEqInteger
-    GHC.IntEqOp   -> mkIntRel PLC.EqInteger
+    GHC.IntQuotOp -> pure $ mkIntFun PLC.DivideInteger
+    GHC.IntRemOp  -> pure $ mkIntFun PLC.RemainderInteger
+    GHC.IntGtOp   -> pure $ mkIntRel PLC.GreaterThanInteger
+    GHC.IntGeOp   -> pure $ mkIntRel PLC.GreaterThanEqInteger
+    GHC.IntLtOp   -> pure $ mkIntRel PLC.LessThanInteger
+    GHC.IntLeOp   -> pure $ mkIntRel PLC.LessThanEqInteger
+    GHC.IntEqOp   -> pure $ mkIntRel PLC.EqInteger
     po            -> throwSd UnsupportedError $ "Primitive operation:" GHC.<+> GHC.ppr po
 
 -- Typeclasses
 
 convEqMethod :: (Converting m) => GHC.Name -> m PLCExpr
 convEqMethod name
-    | name == GHC.eqName = mkIntRel PLC.EqInteger
+    | name == GHC.eqName = pure $ mkIntRel PLC.EqInteger
     | otherwise = throwSd UnsupportedError $ "Eq method:" GHC.<+> GHC.ppr name
 
 convOrdMethod :: (Converting m) => GHC.Name -> m PLCExpr
 convOrdMethod name
     -- only this one has a name defined in the lib??
-    | name == GHC.geName = mkIntRel PLC.GreaterThanEqInteger
-    | GHC.getOccString name == ">" = mkIntRel PLC.GreaterThanInteger
-    | GHC.getOccString name == "<=" = mkIntRel PLC.LessThanEqInteger
-    | GHC.getOccString name == "<" = mkIntRel PLC.LessThanInteger
+    | name == GHC.geName = pure $ mkIntRel PLC.GreaterThanEqInteger
+    | GHC.getOccString name == ">" = pure $ mkIntRel PLC.GreaterThanInteger
+    | GHC.getOccString name == "<=" = pure $ mkIntRel PLC.LessThanEqInteger
```

Comment:
Writing something like

```haskell
case GHC.getOccString name of
    ">"  -> ...
    "<=" -> ...
```

looks slightly nicer to my taste, but is not important.

Reply:
All the cases are a bit special atm - I'll come back to this and the other style suggestions on other bits of code in a follow up.